iceoryx_hoofs/platform/mac/source/pthread.cpp

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+// Copyright (c) 2021 - 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_hoofs/platform/pthread.hpp"
+
+#include <map>
+#include <string>
+#include <utility>
+
+/// @note Since pthread_setname_np and pthread_getname_np are missing in MacOS, their functionality is simulated via a
+/// map of thread handle and thread name.
+static std::map<iox_pthread_t, std::string> handleNameMap;
+
+int iox_pthread_setname_np(iox_pthread_t thread, const char* name)
+{
+    const auto result = handleNameMap.insert(std::make_pair(thread, name));
```

Comment:
This free function can be called concurrently. For instance it is allowed that multiple threads create again multiple threads and set or get the thread name concurrently.
When all those functions are then not guarded by a mutex we have race conditions.

Could you therefore introduce a global `static std::mutex handleNameMapMutex` and always lock it when you access the map with `std::lock_guard<std::mutex> lock(handleNameMapMutex);`

I think the function `iox_pthread_setname_np`, `iox_pthread_getname_np` and `iox_pthread_join` are affected.